### PR TITLE
refactor: new documentation & formatting rules

### PR DESCRIPTION
**Pull request by xnacly**
## Abstract:
This pull request serves the purpose of improving the documentation and the workflow of adding examples, comments, and descriptions by moving away from manually writing HTML documentation to [jsdoc](https://github.com/jsdoc/jsdoc) and generating the documentation using [typedoc](https://typedoc.org/).

#### How to generate the new documentation:
```
npm i -g typedoc
git clone https://github.com/trantlabs/missing-native-js-functions mnjsf
cd mnjsf
typedoc src/*.ts
```
The generated documentation can be found in `./docs`

### Readme:
grafik
#### Screenshot
grafik
#### Corresponding jsdoc function comments:
grafik

## Fixed bugs:
While writing these comments I came across some bugs which I promptly fixed:
- wrong return type for `String.equalsIgnoreCase()`, changed from `string` to `boolean` https://github.com/xNaCly/missing-native-js-functions/commit/741ea9b5cf04633db03898725b6b5ec6f1379729 
- `Date.nowSeconds()` is not a function, constructed Date before calling `.nowSeconds()` on it https://github.com/xNaCly/missing-native-js-functions/commit/7fbaf97338ee8f55538140ef04f54c5caae8ad84
- add missing `timezone: string` function parameter to `DateConstructor.setTimeZone()` https://github.com/xNaCly/missing-native-js-functions/commit/1b44edde000c49759280f2ad15af935458ce152d
- fix typo in `DateConstructor.setTimeZone()` which made the function always return the local date https://github.com/xNaCly/missing-native-js-functions/commit/74b02456f7e45d8a6998718eff3f1c230b860635

## List of changes:
- removed old documentation
- fixed some bugs
- wrote jsdoc and examples for:
  - [x] Array.ts
  - [x] Date.ts
  - [x] Global.ts
  - [x] Math.ts
  - [x] Number.ts
  - [x] Object.ts
  - [x] Promise.ts
  - [x] String.ts
- removed documentation from Readme

## Future todo:
- write tests
- add workflow to build and publish documentation
- add ci

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,20 +6,19 @@
 	Functions</h1>
 <p align="center">A zero-dependecy JavaScript utility library delivering <b>missing</b> native functions 💼</p>
 
-## Why using mnjsf?
+## Why use mnjsf?
 
-mnjsf delivers some functions which are missing in JavaScript
+mnjsf delivers functions missing from JavaScript
```
Comment (Intevel) [inline README.md:11]: Maybe add a `native` before `JavaScript`?

**Review by Intevel — CHANGES_REQUESTED** [review]

**Review by Intevel — APPROVED** [review]

**Review by samuelscheit — APPROVED** [review]

**Comment by xNaCly** [comment]
## Switching Interfaces
Im currently thinking of switching to the `Date` interface, away from the `DateConstructor` interface, due to the fact, that i wasn't able to call the defined functions on the Date without creating an instance.:
image

I also wasn't able to see any doc comments on these functions. It also doesn't make sense to define functions which rely on an instance, such as `setInstance` and `isInPast` on a constructor.

## Fixing bugs
Also, the `DateConstructor.setTimezone` function does not work as intended, it always returns a local date:
```ts
	setTimezone: function (timezone: string) {
		return new Date(
			this.toLocaleString("en-US", {
				timezone,
			})
		);
	},
```
According to the docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/DateTimeFormat#options we should be able to fix this by replacing `timezone` in the options with `timeZone`.

Test in the browser dev tools:
image

@SamuelScheit @Intevel

File: src/Array.ts [hunk 2]
```diff
@@ -1,148 +1,262 @@
...
-    return this.filter((x: any) => arr.includes(x))
-  }
-})
+	remove: function <T>(this: T[], elem: T): T[] {
+		// do not use filter to modify current array
+		const index = this.indexOf(elem);
+		if (index === -1) return this;
+
+		this.splice(index, 1);
+		return this; // .filter((e) => e !== elem);
```
Comment (Intevel) [inline src/Array.ts]: Remove comment

File: src/Object.ts [hunk 3]
```diff
@@ -1,71 +1,112 @@
...
-  merge: function (obj: any) {
-    // this will overwrite if obj has the same property
-    return mergeDeep(obj || {}, this)
-  },
-  stringify: function () {
-    return JSON.stringify(this)
-  }
-})
+		Object.keys(this).forEach((key) => {
+			// @ts-ignore
```
Comment (Intevel) [inline src/Object.ts]: Use ts-expect-error instead of ts-ignore

**Review by Intevel — CHANGES_REQUESTED** [review]

**Review by Intevel — APPROVED** [review]

**Review by samuelscheit — APPROVED** [review]

**Pull request merged**